cmd/mongocli/mongocli.go

```diff
@@ -65,30 +67,40 @@ no additional shell configuration is necessary, see https://docs.brew.sh/Shell-C
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	rootBuilder()
 	if err := rootCmd.Execute(); err != nil {
 		os.Exit(1)
 	}
 }
 
-var (
-	profile string
-)
+// rootBuilder conditionally adds children commands as needed.
+// This is important in particular for Atlas as it dynamically sets flags for cluster creation and
+// this can be slow to timeout on environments with limited internet access (Ops Manager)
+func rootBuilder() {
+	argsWithoutProg := os.Args[1:]
+	if len(argsWithoutProg) != 0 && argsWithoutProg[0] == "--version" {
+		return
+	}
+	rootCmd.AddCommand(cliconfig.Builder())
 
-func init() { //nolint:gochecknoinits // This is the cobra way
+	// We realistically only care about not adding Atlas to the chain of commands
+	// but given we can reuse the pattern for the rest we can save some initialization time here
+	if len(argsWithoutProg) == 0 || argsWithoutProg[0] == atlas.Use {
+		rootCmd.AddCommand(atlas.Builder())
```

Comment:
[q] how do you know that the first argument is the service? This question may be related to the fact that I don't know what `[1:]` returns.

Reply:
if you call the cli like `mongocli atlas clusters` then `os.Args` is `["mongocli", "atlas", "clusters"]`, `os.Args[1:]` would be `["atlas", "clusters"]`

Reply:
thanks!!!